src/content/config.ts

```diff
@@ -5,7 +5,12 @@ const documents = defineCollection({
     schema: zod.object({
         title: zod.string(),
         search: zod.boolean().default(true),
-        author: zod.string().default("Anonymous"),
+
+        author: zod.string().optional(),
+
+        author_data: zod.record(zod.unknown()).optional(),
```

Comment:
## TL;DR: `config.ts` (bad name by Astro, will look into) defines properties in MDX files. `author_data` isn't shown to be used anywhere but `src/pages/[...slug].astro`, so it should be a type in that file or a different file.

Admittedly, `config.ts` isn't the most descriptive name that Astro has picked. This file determines what you an expect in MDX files within [`src/content/documents`](https://github.com/haotian2006/MyDocs/tree/Integrate-DiscordAPI/src/content/documents), so `author_data` which is not defined in other MDX files and is exclusively used in `src/pages/[...slug].astro` should be a type and either be placed there or shoved into a separate file altogether. I'll see if we can't choose a more apt name separately.